firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/SettingsCache.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions.settings
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.doublePreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.core.longPreferencesKey
+import kotlinx.coroutines.flow.first
+
+internal data class SessionConfigs(
+  val sessionEnabled: Boolean?,
+  val sessionSamplingRate: Double?,
+  val cacheDuration: Int?,
+  val sessionRestartTimeout: Int?,
+  val cacheUpdatedTime: Long?
+)
+
+internal class SettingsCache(private val store: DataStore<Preferences>) {
+  private var sessionConfigs =
+    SessionConfigs(
+      sessionEnabled = null,
```

Comment:
If you put the default ` = null` on each property when you define the data class, this default instance can be just one line. That lets you write out each property name one fewer times.

Reply:
Like the succinctness. Applied it.